plugins/pipelines/badge.js

```diff
@@ -9,23 +9,36 @@ const idSchema = joi.reach(schema.models.pipeline.base, 'id');
  * Generate Badge URL
  * @method getUrl
  * @param  {string}  badgeService    Template URL for badges - needs {{status}} and {{color}}
- * @param  {string}  [currentStatus] Current Build Status
+ * @param  {Array}  [buildsStatus]   Current status of all builds in the same event
  * @return {string}                  URL to redirect to
  */
-function getUrl(badgeService, currentStatus) {
-    const status = (currentStatus || 'UNKNOWN').toLowerCase();
+function getUrl(badgeService, buildsStatus = []) {
+    const counts = {};
+    const parts = [];
+    let worst = 'lightgrey';
+
     const statusColor = {
-        unknown: 'lightgrey',
         success: 'green',
-        failure: 'red',
-        aborted: 'red',
         queued: 'blue',
-        running: 'blue'
+        running: 'blue',
+        unknown: 'lightgrey',
+        failure: 'red',
+        aborted: 'red'
     };
-    const color = statusColor[status];
+
+    buildsStatus.forEach((status) => {
+        counts[status] = (counts[status] || 0) + 1;
+    });
+
+    Object.keys(statusColor).forEach((s) => {
+        if (counts[s]) {
+            parts.push(`${counts[s]} ${s}`);
+            worst = statusColor[s];
```

Comment:
An Ojbect is unordered, so you can't necessarily rely on that. If you need order, consider using a map https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared.